lighthouse-core/test/audits/timing-budget-test.js

```diff
@@ -0,0 +1,201 @@
+const devtoolsLog = require('../fixtures/traces/progressive-app-m60.devtools.log.json');
+
+/* eslint-env jest */
+
+describe('Performance: Timing budget audit', () => {
+  let artifacts;
+  let context;
+  beforeEach(() => {
+    artifacts = {
+      devtoolsLogs: {
+        defaultPass: devtoolsLog,
+      },
+      traces: {defaultPass: trace},
+      URL: {requestedUrl: 'http://example.com', finalUrl: 'http://example.com'},
+    };
+
+    context = {
+      computedCache: new Map(),
+      settings: {
+        throttlingMethod: 'simulate',
+      },
+    };
+  });
+
+  describe('with a budget.json', () => {
+    beforeEach(() => {
+      context.settings.budgets = [{
+        path: '/',
+        timings: [
+          {
+            metric: 'first-contentful-paint',
+            budget: 9999,
+          },
+          {
+            metric: 'estimated-input-latency',
+            budget: 10,
+          },
+        ],
+      }];
+    });
+
+    it('table headers are correct', async () => {
+      const result = await TimingBudgetAudit.audit(artifacts, context);
+      const headers = result.details.headings;
+      expect(headers[0].text).toBeDisplayString('Timing Metric');
+      expect(headers[1].text).toBeDisplayString('Duration');
+      expect(headers[2].text).toBeDisplayString('Over Budget');
+    });
+
+    it('table item info is correct', async () => {
+      const result = await TimingBudgetAudit.audit(artifacts, context);
+      const items = result.details.items;
+      // Failing Budget
+      expect(items[0].label).toBeDisplayString('Estimated Input Latency');
+      expect(items[0].measurement).toBeCloseTo(77.8);
+      expect(items[0].overBudget).toBeCloseTo(67.8);
+    });
+
+    it('calculates the "overBudget" column correctly', async () => {
+      const result = await TimingBudgetAudit.audit(artifacts, context);
+
+      // Failing Budget
+      expect(result.details.items[0].overBudget).toBeCloseTo(67.8);
+
+      // Passing Budget
+      expect(result.details.items[1].overBudget).toBeUndefined();
+    });
+
+    it('only includes rows for timing metrics with budgets', async () => {
+      const result = await TimingBudgetAudit.audit(artifacts, context);
+      expect(result.details.items).toHaveLength(2);
+    });
+
+    it('works for all supported timing metrics', async () => {
+      const metrics = [
+        'first-contentful-paint',
+        'first-cpu-idle',
+        'interactive',
+        'first-meaningful-paint',
+        'max-potential-fid',
+        'estimated-input-latency',
+        'total-blocking-time',
+        'speed-index',
+      ];
+      metrics.forEach(async (metric) => {
+        context.settings.budgets = [{
+          path: '/',
+          timings: [
+            {
+              metric: metric,
+              budget: 100,
+            },
+          ],
+        }];
+        const result = await TimingBudgetAudit.audit(artifacts, context);
+        expect(result).toBeDefined();
```

Comment:
is this really just trying to assert that it doesn't throw? maybe we could have it check `expect(result.details.items).toHaveLength(1)`?